uiTests/src/test/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala

```diff
@@ -2,18 +2,40 @@ package com.virtuslab.gitmachete.uitest
 
 import java.nio.file.Files
 import java.nio.file.attribute.FileTime
-
 import com.virtuslab.gitmachete.testcommon.BaseGitRepositoryBackedIntegrationTestSuite
 import com.virtuslab.gitmachete.testcommon.BaseGitRepositoryBackedIntegrationTestSuite.SETUP_WITH_SINGLE_REMOTE
-import org.junit.{After, Assert, Before, Test}
+import org.junit.{ After, AfterClass, Assert, Before, BeforeClass, Test }
 import org.junit.runner.RunWith
 import org.junit.runners.JUnit4
 import org.virtuslab.ideprobe.Extensions._
 import org.virtuslab.ideprobe._
 import org.virtuslab.ideprobe.config._
 import org.virtuslab.ideprobe.dependencies._
 import org.virtuslab.ideprobe.ide.intellij.IntelliJFactory
-import org.virtuslab.ideprobe.junit4.RunningIntelliJPerSuite
+
+trait RunningIntelliJPerSuite extends RunningIntelliJPerSuiteBase {
+  @BeforeClass override final def setup(): Unit = super.setup()
+
+  @AfterClass override final def teardown(): Unit = {
+    try {
+      super.teardown()
+    } catch {
+      case e: Exception =>
```

Comment:
As I'm not familiar with Scala I will ask why do we have to write singe case here? 
Does it allow us for some type interference or is it not to catch Throwable?
Is it some kind of Scala convention to prefer single case over if?

Reply:
It's just that I checked the impl of `super.teardown`, and they simply always throw a `new Exception` in case of any IDE errors (yeah lol xD)

Reply:
> Is it some kind of Scala convention to prefer single case over if?

I'd say that yes, Scala likes pattern matching in general (even in cases where `if-else` could be equivalently used), and even requires pattern matching in certain cases &mdash; most notably for `catch` clause, even when just catching a single exception type

Reply:
That's just the equivalent of Java's:
```java
try {
...
} catch (Exception e) {
...
}
```